src/ci_providers/provider_codebuild.js

```diff
@@ -0,0 +1,83 @@
+function detect(envs) {
+  return !!envs.CODEBUILD_CI
+}
+
+function _getBuild(inputs) {
+  const { args, envs } = inputs
+  return args.build || envs.CODEBUILD_BUILD_ID
+}
+
+// eslint-disable-next-line no-unused-vars
+function _getBuildURL(inputs) {
+  return ''
+}
+
+function _getBranch(inputs) {
+  const { args, envs } = inputs
+  return (
+    args.branch ||
+    (envs.CODEBUILD_WEBHOOK_HEAD_REF
+      ? envs.CODEBUILD_WEBHOOK_HEAD_REF.replace(/^refs\/heads\//, '')
+      : '')
+  )
+}
+
+function _getJob(envs) {
+  return envs.CODEBUILD_BUILD_ID
+}
+
+function _getPR(inputs) {
+  const { args, envs } = inputs
+  return (
+    args.pr ||
+    (envs.CODEBUILD_WEBHOOK_HEAD_REF &&
+    envs.CODEBUILD_SOURCE_VERSION.startsWith('pr/')
+      ? envs.CODEBUILD_SOURCE_VERSION.replace(/^pr\//, '')
+      : '')
+  )
+}
+
+function _getService() {
+  return 'codebuild'
+}
+
+function getServiceName() {
+  return 'AWS CodeBuild'
+}
+
+function _getSHA(inputs) {
+  const { args, envs } = inputs
+  return args.sha || envs.CODEBUILD_RESOLVED_SOURCE_VERSION || ''
+}
+
+function _getSlug(inputs) {
+  const { args, envs } = inputs
+  return (
+    args.slug ||
+    (envs.CODEBUILD_SOURCE_REPO_URL
+      ? envs.CODEBUILD_SOURCE_REPO_URL.replace(/^.*github.com\//, '').replace(
+          /\.git$/,
+          '',
+        )
```

Comment:
This will only work for GitHub repositories. In other words, AWS repositories will not work.

Reply:
Yeah this was a direct copy from the node repo. We could change it to something else

Reply:
I *just* realized that. Since most of the functionality over there is being ported over here, are there explicit plans to deprecate that project?

Reply:
I think so.

Reply:
> This will only work for GitHub repositories. In other words, AWS repositories will not work.

AWS repositories are not supported by Codecov, we have no system on place to retrieve code from them.

We support GitHub, Bitbucket , and GitLab only.